yb-voyager/src/tgtdb/oracle.go

```diff
@@ -299,43 +301,77 @@ func (tdb *TargetOracleDB) getLiveMigrationMetaInfoByTable(conn *sql.Conn, migra
 
 func (tdb *TargetOracleDB) initChannelMetaInfo(conn *sql.Conn, migrationUUID uuid.UUID, numChans int) error {
 	// if there are >0 rows, then skip because already been inited.
-	rowCount, err := tdb.getLiveMigrationMetaInfo(conn, migrationUUID)
+	rowCount, err := tdb.getEventChannelsRowCount(conn, migrationUUID)
 	if err != nil {
 		return fmt.Errorf("error getting channels meta info for %s: %w", EVENT_CHANNELS_METADATA_TABLE_NAME, err)
 	}
 	if rowCount > 0 {
 		log.Info("event channels meta info already created. Skipping init.")
 		return nil
 	}
+	ctx := context.Background()
+	tx, err := conn.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("error creating tx: %w", err)
+	}
+	defer tx.Rollback()
 	for c := 0; c < numChans; c++ {
 		insertStmt := fmt.Sprintf("INSERT INTO %s VALUES ('%s', %d, -1, %d, %d, %d)", EVENT_CHANNELS_METADATA_TABLE_NAME, migrationUUID, c, 0, 0, 0)
-		_, err := conn.ExecContext(context.Background(), insertStmt)
+		_, err := tx.Exec(insertStmt)
 		if err != nil {
 			return fmt.Errorf("error executing stmt - %v: %w", insertStmt, err)
 		}
 		log.Infof("created channels meta info: %s;", insertStmt)
 	}
+	err = tx.Commit()
+	if err != nil {
+		return fmt.Errorf("error committing tx: %w", err)
+	}
 	return nil
 }
 
-func (tdb *TargetOracleDB) initEventStatByTable(conn *sql.Conn, migrationUUID uuid.UUID, tableName string) error {
-	insertStmt := fmt.Sprintf("INSERT INTO %s VALUES ('%s', '%s', %d, %d, %d, %d)", EVENTS_PER_TABLE_METADATA_TABLE_NAME, migrationUUID, tableName, 0, 0, 0, 0)
-	_, err := conn.ExecContext(context.Background(), insertStmt)
+func (tdb *TargetOracleDB) initEventStatByTableMetainfo(tableNames []string, migrationUUID uuid.UUID, conn *sql.Conn, numChans int) error {
+
+	ctx := context.Background()
+	tx, err := conn.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("error creating tx: %w", err)
+	}
+	defer tx.Rollback()
+	for _, tableName := range tableNames {
+		for c := 0; c < numChans; c++ {
+			tableName = tdb.qualifyTableName(tableName)
+			rowCount, err := tdb.getLiveMigrationMetaInfoByTable(conn, migrationUUID, tableName, c)
+			if err != nil {
+				return  fmt.Errorf("failed to get table wise meta info: %w", err)
+			}
+			if rowCount > 0 {
+				log.Info("table wise meta info already created. Skipping init.")
+			} else {
```

Comment:
Shouldn't this be outside of the `numChans` loop?

Reply:
which one, `rowCount` check?